src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -966,7 +966,7 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						}
 						foreach (var typeNameValue in methodParams[0].UniqueValues ()) {
 							if (typeNameValue is KnownStringValue knownStringValue) {
-								TypeReference foundTypeRef = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents);
+								TypeReference foundTypeRef = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents, callingMethodDefinition);
```

Comment:
IMHO this should be the only place that is allowed to call ResolveTypeName with an unqualified assembly. All the other callers of ResolveTypeName should require the string to be assembly qualified and warn if it isn't. https://github.com/mono/linker/issues/1895#issuecomment-800062109